### PR TITLE
refactor(docs): code block demo as markdownit plugin

### PR DESCRIPTION
**Pull request by yinonov**

### CODE REVIEW

File: libs/components/src/lib/accordion-item/README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ Use the `open` attribute to set the accordion-item's open state.
 - Type: `boolean`
 - Default: `false`
 
-```html preview
+```html-preview
```
Comment (yinonov) [inline libs/components/src/lib/accordion-item/README.md:53]:
```suggestion
```html preview
```

File: apps/docs/transformers/code-block-demo.js [hunk 2]
```diff
@@ -69,74 +68,70 @@ const getHtml = (demoData) => {
...
+      </div>
+    </vwc-elevation>`;
 }
 
 const getIframe = (frameData) => {
-    const saveFolder = verifyAndCreateSaveFolder(frameData.outputPath);
-    frameData.saveFolder = saveFolder;
-    const filePath = saveCodeAsHTMLFile(frameData);
-    return filePath.substring(saveFolder.indexOf('docs/') + 4);
+    try {
```
Comment (rinaok) [inline apps/docs/transformers/code-block-demo.js:76]: All the try-and-catch is no longer necessary. I added them for debugging purposes.

**Comment by rinaok** [comment]
@yinonov Looks very nice!

File: apps/docs/transformers/code-block-demo.js [hunk 3]
```diff
@@ -69,74 +68,70 @@ const getHtml = (demoData) => {
...
+        const saveFolder = verifyAndCreateSaveFolder(frameData.outputPath);
+        frameData.saveFolder = saveFolder;
+        const filePath = saveCodeAsHTMLFile(frameData);
+        return filePath.substring(saveFolder.indexOf('docs/') + 4);
+    }
+    catch (e) {
+        console.log("getIframe error: ", e);
+    }
```
Comment (rinaok) [inline apps/docs/transformers/code-block-demo.js:84]: All the try-and-catch is no longer necessary. I added them for debugging purposes.

File: apps/docs/transformers/code-block-demo.js [hunk 4]
```diff
@@ -69,74 +68,70 @@ const getHtml = (demoData) => {
...
-    if (!fs.existsSync(saveFolder)) {
-      fs.mkdirSync(saveFolder, { recursive: true });
+    try {
+        const saveFolder = path.join(path.dirname(outputPath), '/frames');
+        if (!fs.existsSync(saveFolder)) {
+            fs.mkdirSync(saveFolder, { recursive: true });
+        }
+        return saveFolder;
+    } catch (e) {
+        console.log("verifyAndCreateSaveFolder error: ", e);
```
Comment (rinaok) [inline apps/docs/transformers/code-block-demo.js:95]: Also here

File: apps/docs/transformers/code-block-demo.js [hunk 5]
```diff
@@ -69,74 +68,70 @@ const getHtml = (demoData) => {
...
+    try {
+        const filePath = `${frameData.saveFolder}/${frameData.codeBlockId}.html`;
+        const componentName = getComponentName(frameData.outputPath);
+        frameData.demoStr += addModules(componentName);
+        fs.writeFileSync(filePath, frameData.demoStr);
+        return filePath;
+    }
+    catch (e) {
+        console.log("saveCodeAsHTMLFile error: ", e);
+    }
```
Comment (rinaok) [inline apps/docs/transformers/code-block-demo.js:109]: And here

File: apps/docs/transformers/code-block-demo.js [hunk 6]
```diff
@@ -69,74 +68,70 @@ const getHtml = (demoData) => {
...
-    const pathName = path.dirname(outputPath).substring(0, outputPath.lastIndexOf('/'));
-    const componentName = pathName.substring(pathName.lastIndexOf('/') + 1);
-    return componentName;
+    try {
+        const pathName = path.dirname(outputPath).substring(0, outputPath.lastIndexOf('/'));
+        const componentName = pathName.substring(pathName.lastIndexOf('/') + 1);
+        return componentName;
+    } catch (e) {
+        console.log("getComponentName error: ", e);
+    }
```
Comment (rinaok) [inline apps/docs/transformers/code-block-demo.js:119]: .

File: apps/docs/transformers/code-block-demo.js [hunk 7]
```diff
@@ -69,74 +68,70 @@ const getHtml = (demoData) => {
...
+    try {
+        let modulesStr = '';
+        let component = jsonData.filter(item => item.title.includes(componentName));
+        component[0].modules.forEach(module => {
+            modulesStr += `<script type="module" src="${module}"></script>`;
+        });
+        return modulesStr;
+    } catch (e) {
+        console.log("addModules error: ", e);
+    }
```
Comment (rinaok) [inline apps/docs/transformers/code-block-demo.js:132]: .

**Review by rinaok — CHANGES_REQUESTED** [review]

**Comment by rinaok** [comment]
Screen Shot 2022-05-23 at 15 43 38

Is this the way it should be? Without highlighting?

File: apps/docs/transformers/code-block-demo.js (hunk 3 above)
Comment (yinonov) [reply apps/docs/transformers/code-block-demo.js:84]: oh 🫤

**Comment by yinonov** [comment]
> Screen Shot 2022-05-23 at 15 43 38
> 
> Is this the way it should be? Without highlighting?

no. strange that's what u get
it uses our tokens now

image

image

**Comment by rinaok** [comment]
> > Screen Shot 2022-05-23 at 15 43 38
> > Is this the way it should be? Without highlighting?
> 
> no. strange that's what u get it uses our tokens now
> 
> image
> 
> image

@yinonov 

Look at accordion-item: https://vivid-ft84re92m1n0.deno.dev/components/accordion-item

**Pull request merged**